src/Helper/Normalizer.php

```diff
@@ -106,13 +105,21 @@ public static function renderLabel(
     ): string {
         $html = '';
 
+        $tagName = self::iconTagName($iconAttributes);
+
+        unset($iconAttributes['tagName']);
+
         if ($iconClass !== '') {
             Html::addCssClass($iconAttributes, $iconClass);
         }
 
         if ($icon !== '' || $iconAttributes !== [] || $iconClass !== '') {
-            $i = I::tag()->attributes($iconAttributes)->content($icon);
-            $html = Span::tag()->attributes($iconContainerAttributes)->content($i)->encode(false)->render();
+            $tag = Html::tag($tagName)->attributes($iconAttributes)->content($icon);
+
+            $html = match ($tagName) {
+                'i' => Span::tag()->attributes($iconContainerAttributes)->content($tag)->encode(false)->render(),
+                default => $tag->render(),
+            };
```

Comment:
A simple `if` would be better here because we have no other special cases.

Reply:
done.